tools/ktx/command_extract.cpp

```diff
@@ -296,14 +235,57 @@ void CommandExtract::OptionsExtract::process(cxxopts::Options&, cxxopts::ParseRe
             report.fatal_usage("Conflicting options: --layer cannot be used with --all.");
         if (face)
             report.fatal_usage("Conflicting options: --face cannot be used with --all.");
-        if (depth)
+        if (depth_)
             report.fatal_usage("Conflicting options: --depth cannot be used with --all.");
 
         level = all;
         layer = all;
         face = all;
-        depth = all;
+        depth_ = all;
     }
+
+    if (globalAll && outputPath == "-")
+        report.fatal_usage("stdout cannot be used with multi-output '--all' extract.");
+    if (level == all && outputPath == "-")
+        report.fatal_usage("stdout cannot be used with multi-output '--level all' extract.");
+    if (layer == all && outputPath == "-")
+        report.fatal_usage("stdout cannot be used with multi-output '--layer all' extract.");
+    if (face == all && outputPath == "-")
+        report.fatal_usage("stdout cannot be used with multi-output '--face all' extract.");
+    if (depth_ == all && outputPath == "-")
+        report.fatal_usage("stdout cannot be used with multi-output '--depth all' extract.");
+
+    if (args["uri"].count()) {
+        uriFlagUsed = true;
+        try {
+            fragmentURI = parseFragmentURI(args["uri"].as<std::string>());
+        } catch (const std::exception& e) {
+            report.fatal_usage("Failed to parse Fragment URI: {}", e.what());
+        }
+        if (globalAll)
```

Comment:
Why not do these ifs, up to line 272,  before parsing the fragment URI?

Reply:
Yeah, It would make more sense. I have reordered them.